### ISSUE TITLE
General Prefix Greater Than 1 Character Breaks Parsing

### ISSUE THREAD

**Issue by bonedaddy** [zekroTJA/shireikan#1]
# Overview

Hello I'm switching to this package from `dgc` where I was using a prefix that was larger than one character (`!coomer`) and parsing of commands was working correctly. However attempting to use the same pattern with shireikan causes a "command not found" error when attempting to parse a command. Changing from `!coomer` to `!` works, however even using `!c` causes the error to happen again.

It would be a good feature to have if we could have longer than 1 character prefixes, as well as the ability to specify alternate prefixes.

# Reproduction

* Set prefix to greater than 1 character (ie `!coomer`)
* Try to run a command like `!coomer help` and observe error

**Comment by zekroTJA** [comment]
The problem is, that shireikan can not handle prefixes with spaces. 

For better understanding, take a look at the parsing of the prefix:

```
!ping hey
^^    ^
||    |
||    +-- Argument
|+------- Invoke
+-------- Prefix
```

Using prefixes with multiple characters is possible, for example, the default prefix of my bot shinpuru which is `sp!`:
```
sp!ping hey
^  ^    ^
|  |    |
|  |    +-- Argument
|  +------- Invoke
+---------- Prefix
```

Following this structure, you would need to invoke commands as following:
```
!coomerping hey
^      ^    ^
|      |    |
|      |    +-- Argument
|      +------- Invoke
+-------------- Prefix
```

So, you would need to define your prefix with a leading space: `!coomer `.

The thing is, that would actually panic when parsing a command message. I've fixed this in the following commit: 4a066f2f293f2af240226a6c9208568b2f949f2b.

Maybe I'll consider to implement an option to allow white spaces between prefix and invoke, but that would need further restructuring of the command handler.

**Comment by zekroTJA** [comment]
If this does not fix the issue in your use case, I'll re-open the issue. ^^
